backend/de.metas.async/src/main/java/de/metas/async/api/impl/AsyncBatchBL.java

```diff
@@ -213,145 +171,92 @@ public void enqueueAsyncBatch(@NonNull final AsyncBatchId asyncBatchId)
 
 		//
 		final Timestamp firstEnqueued = asyncBatch.getFirstEnqueued();
 		if (firstEnqueued == null)
 		{
 			// shall not happen
-			return false;
+			return Optional.of(processedTimeOffsetMillis);
 		}
 
 		//
 		final Timestamp lastEnqueued = asyncBatch.getLastEnqueued();
 		if (lastEnqueued == null)
 		{
 			// shall not happen
-			return false;
+			return Optional.of(processedTimeOffsetMillis);
 		}
 
 		final Timestamp lastProcessed = asyncBatch.getLastProcessed();
 		if (lastProcessed == null)
 		{
 			// shall not happen
-			return false;
+			return Optional.of(processedTimeOffsetMillis);
 		}
 
 		// Case: when did not pass enough time between fist enqueue time and now
-		final int processedTimeOffsetMillis = getProcessedTimeOffsetMillis();
 		final Timestamp now = de.metas.common.util.time.SystemTime.asTimestamp();
-		final Timestamp minTimeAfterFirstEnqueued = TimeUtil.addMillis(now, processedTimeOffsetMillis);
-		if (firstEnqueued.compareTo(minTimeAfterFirstEnqueued) > 0)
-		{
-			return false;
-		}
 
-		// Case: when last processed time is before last enqueued time; this means that we still have packages to process
-		if (lastProcessed.compareTo(lastEnqueued) < 0)
+		final Timestamp minTimeAfterFirstEnqueued = TimeUtil.addMillis(firstEnqueued, processedTimeOffsetMillis);
+
+		if (minTimeAfterFirstEnqueued.compareTo(now) > 0)
 		{
-			return false;
+			final int millisToWait = TimeUtil.getMillisBetween(now, minTimeAfterFirstEnqueued);
+
+			return Optional.of(millisToWait);
 		}
 
 		// Case: when did not pass enough time between last processed time and now - offset
 		// take a bigger time for checking processed because thread could be locked by other thread and we could have some bigger delay
-		final Timestamp minTimeAfterLastProcessed = TimeUtil.addMillis(now, processedTimeOffsetMillis);
-		if (lastProcessed.compareTo(minTimeAfterLastProcessed) > 0)
+		final Timestamp minTimeAfterLastProcessed = TimeUtil.addMillis(lastProcessed, processedTimeOffsetMillis);
+		if (minTimeAfterLastProcessed.compareTo(now) > 0)
 		{
-			return false;
+			final int millisToWait = TimeUtil.getMillisBetween(now, minTimeAfterLastProcessed);
+
+			return Optional.of(millisToWait);
 		}
 
 		//
-		// If we reach this point, our batch can be considered processed
-		return true;
-	}
-
-	/**
-	 * assumes that asyncBatch.isProcessed() was already set with the help of #checkProcessed 
-	 */
-	private boolean checkProcessing(@NonNull final I_C_Async_Batch asyncBatch)
-	{
-		if (asyncBatch.isProcessed())
-		{
-			return false;
-		}
-		final int countEnqueued = asyncBatch.getCountEnqueued();
-		return countEnqueued > 0;
-	}
-	
-	private int getProcessedTimeOffsetMillis()
-	{
-		return Services.get(ISysConfigBL.class).getIntValue("de.metas.async.api.impl.AsyncBatchBL_ProcessedOffsetMillis", 1);
+		// If we reach this point, we can move on and check if the async batch is processed
+		return Optional.empty();
```

Comment:
so i guess, instead of having an extra Optional, we could also return `Duration.ZERO`. i think that would make things clearer